### ISSUE TITLE
ObjectDirective does not work with defineSlots after upgrading to 2.0.15

### ISSUE THREAD

**Issue by skywalker512** [vuejs/language-tools#4327]
When both ObjectDirective and defineSlots appear in the same sfc, ObjectDirective cannot work (v-test must be written in the slot).

```vue
<script setup lang="ts">
import { vTest } from '../directive';

defineSlots<{
  default(): any;
}>();
</script>

<template>
  <h1>{{ msg }}</h1>

  <slot>
    <p v-test="{}">Default slot content</p>
  </slot>
</template>
```

Reproduce repo https://github.com/skywalker512/vue-ls-directive

**Comment by so1ve** [comment]
Maybe fixed by https://github.com/vuejs/language-tools/pull/4326? Could you please check @zhiyuanzmj?

**Comment by zhiyuanzmj** [comment]
> Maybe fixed by #4326? Could you please check @zhiyuanzmj?

No, It only affects the slots destructuring
